internal/controllers/objectsets/remotephase_reconciler.go

```diff
@@ -51,7 +52,7 @@ func (r *objectSetRemotePhaseReconciler) Teardown(
 	ctx context.Context, objectSet adapters.ObjectSetAccessor,
 	phase corev1alpha1.ObjectSetTemplatePhase,
 ) (cleanupDone bool, err error) {
-	log := logr.FromContextOrDiscard(ctx)
+	log := ctrl.LoggerFrom(ctx)
```

Comment:
ctrl.LoggerFrom does a silent fallback to a default logger in https://github.com/kubernetes-sigs/controller-runtime/blob/4c2e9cec2954e6d5a9522c65b2e22f24acb750b5/pkg/log/log.go#L87

Is this what we want here?

Reply:
From my understanding:
1. Using `ctrl.LoggerFrom(ctx)` aligns with our requirement to avoid discarding logs. It ensures that even if a logger is missing from the context, the logs will still be directed to the global controller-runtime logger.
2. In my opinion, this is the safest way to avoid discarding logs. It is very hard to trace back every `ctx` whether there is a logger injected, even our unit test and integration test are all passed, it still cannot guarantee every `ctx` we used has a logger, and it can be very risky and lead to breaking change if we only use `logr.FromContext(ctx)` and return panic